Simplenote/src/androidTest/java/com/automattic/simplenote/screenshots/ScreenshotTest.java

```diff
@@ -1,58 +1,527 @@
+                .setClock("1231")
+                .setWifiVisibility(SHOW)
+                .setBarsMode(TRANSPARENT)
+                .enable();
+    }
+
     @Test
-    public void screenshotTest() {
-        ViewInteraction loginButton = onView(
-                allOf(ViewMatchers.withId(R.id.button_login), isDisplayed())
-        );
-        loginButton.perform(click());
-
-        ViewInteraction loginWithEmailButton = onView(
-                allOf(withId(R.id.button_email), isDisplayed())
-        );
-        loginWithEmailButton.perform(click());
-
-        Screengrab.screenshot("test-screenshot");
+    public void screenshotTest() throws InterruptedException {
+        // Pre-checks if the state is dirty
+        enterThenDisablePasscodeIfNeeded();
+        logoutIfNeeded();
+
+        login();
+
+        // Wait for notes to load.
+        // Not 100% sure why, but without this little sleep call, the call to wait for the note that
+        // we need to interact with to be loaded in the list fails. I suspect it might be due to the
+        // app changing root activity (from login to notes list) but I haven't had the time to
+        // research it properly
+        Thread.sleep(2000);
+        waitForViewMatching(allOf(withId(R.id.note_title), withText(NOTE_TITLE)), 5000);
+
+        selectNoteFromNotesList();
+
+        Screengrab.screenshot("note");
+
+        dismissNoteEditor();
+
+        loadSearchFromNotesList("Recipe");
+        // Make sure the results have been rendered
+        waitForViewMatching(allOf(withId(R.id.note_title), withText(NOTE_TITLE)), 1000);
+
+        Screengrab.screenshot("search");
+
+        dismissSearch();
+
+        enableDarkModeFromNotesList();
+
+        dismissSettings();
+
+        Screengrab.screenshot("notes-list");
+
+        loadSideMenuFromNotesList();
+
+        Screengrab.screenshot("tags");
+
+        loadSettingsFromSideMenu();
+
+        loadPasscodeSetterFromSettings();
+
+        // Set
+        typeFullPasscode();
+        // Confirm
+        typeFullPasscode();
+
+        // Relaunch Simplenote to show the pin screen
+        relaunchSimplenote();
+
+        // We want only 3 numbers in the passcode screenshot. Also, as soon as we press the fourth,
+        // the screen is dismissed, so we wouldn't be able to take the screenshot.
+        tapPasscodeKeypad();
+        tapPasscodeKeypad();
+        tapPasscodeKeypad();
+        Screengrab.screenshot("pin");
+        tapPasscodeKeypad();
+
+        loadSettingsFromNotesList();
+        // Disable passcode
+        loadPasscodeUnsetterFromSettings();
+        typeFullPasscode();
+        // Disable darkmode
+        disableDarkModeFromSettings();
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        CleanStatusBar.disable();
+    }
+
+    // Flows
+
+    private void logoutIfNeeded() throws InterruptedException {
+        if (isViewDisplayed(getViewById(R.id.list_root)) == false) {
```

Comment:
Sure. Is this the convention you use in the codebase?

For context. I tend to use this style because I find it clearer to read. I sometimes miss the `!` at the start, just a single thin character... the more verbose `== false` seems clearer.

Reply:
Oh... I just saw that Android Studio offers that as an option. Cool, I guess that's just the expected style here. 

Applying 🔨